m4/boost.m4

```diff
@@ -496,7 +496,7 @@ dnl generated only once above (before we start the for loops).
       if test x"$Boost_lib" = xyes; then
         # Check or used cached result of whether or not using -R or
         # -rpath makes sense.  Some implementations of ld, such as for
-        # Mac OSX, require -rpath but -R is the flag known to work on
+        # Mac OS X, require -rpath but -R is the flag known to work on
```

Comment:
I personally favor `macOS` as that's been current for quite a while, but it's possible this file is really talking about older generations, so I went w/ the simpler/older fix.

I'm not sure if this file is a third party file (it probably is) or if I've already sent them a PR (quite possibly)

Reply:
It is a third-party file originally, yes